Makefile

```diff
@@ -1,110 +1,154 @@
 VERSION=$(shell cat VERSION | tr -d " \t\n\r")
-SRC=$(shell find . -type f -name '*.go') pkg/manifests/bindata.go
+GO111MODULE?=off
+export GO111MODULE
+
+PKGS=$(shell go list ./... | grep -v -E '/vendor/|/test|/examples')
+GOLANG_FILES:=$(shell find . -name \*.go -print) pkg/manifests/bindata.go
 FIRST_GOPATH:=$(firstword $(subst :, ,$(shell go env GOPATH)))
 EMBEDMD_BIN=$(FIRST_GOPATH)/bin/embedmd
 GOBINDATA_BIN=$(FIRST_GOPATH)/bin/go-bindata
-GOJSONTOYAML_BIN=$(FIRST_GOPATH)/bin/gojsontoyaml
-# We need jsonnet on Travis; here we default to the user's installed jsonnet binary; if nothing is installed, then install go-jsonnet.
-JSONNET_BIN=$(if $(shell which jsonnet 2>/dev/null),$(shell which jsonnet 2>/dev/null),$(FIRST_GOPATH)/bin/jsonnet)
-JB_BIN=$(FIRST_GOPATH)/bin/jb
+JB_BINARY=$(FIRST_GOPATH)/bin/jb
+GOJSONTOYAML_BINARY=$(FIRST_GOPATH)/bin/gojsontoyaml
 ASSETS=$(shell grep -oh 'assets/.*\.yaml' pkg/manifests/manifests.go)
 JSONNET_SRC=$(shell find ./jsonnet -type f)
 JSONNET_VENDOR=jsonnet/jsonnetfile.lock.json jsonnet/vendor
-GO_BUILD_RECIPE=GOOS=$(GOOS) go build --ldflags="-s -X github.com/openshift/cluster-monitoring-operator/pkg/operator.Version=$(VERSION)" -o $(BIN) $(MAIN_PKG)
 
-build: $(BIN)
+GO_BUILD_RECIPE=GOOS=linux CGO_ENABLED=0 go build --ldflags="-s -X $(GO_PKG)/pkg/operator.Version=$(VERSION)"
+CONTAINER_CMD:=docker run --rm \
+		-u="$(shell id -u):$(shell id -g)" \
+		-v "$(shell go env GOCACHE):/.cache/go-build" \
+		-v "$(PWD):/go/src/$(GO_PKG):Z" \
+		-w "/go/src/$(GO_PKG)" \
+		-e GO111MODULE=$(GO111MODULE) \
+		quay.io/coreos/jsonnet-ci
+
+.PHONY: all
+all: format generate build test
+
+.PHONY: clean
+clean:
+	# Remove all files and directories ignored by git.
+	git clean -Xfd .
+
+############
+# Building #
+############
+
+.PHONY: build-in-docker
+build-in-docker:
+	$(CONTAINER_CMD) $(MAKE) $(MFLAGS) build
 
-$(BIN): $(SRC)
-	$(GO_BUILD_RECIPE)
+.PHONY: build
+build: operator
+
+.PHONY: operator
+operator: $(GOLANG_FILES)
+	$(GO_BUILD_RECIPE) -o operator $(GO_PKG)/cmd/operator
 
 # We need this Make target so that we can build the operator depending
 # only on what is checked into the repo, without calling to the internet.
+.PHONY: operator-no-deps
 operator-no-deps:
-	$(GO_BUILD_RECIPE)
-
-run: build
-	./$(BIN)
+	$(GO_BUILD_RECIPE) -o operator $(GO_PKG)/cmd/operator
 
-crossbuild:
-	$(ENVVAR) $(MAKE) build
+.PHONY: image
+image: .hack-operator-image
 
-container:
+.hack-operator-image: Dockerfile
+# Create empty target file, for the sole purpose of recording when this target
+# was last executed via the last-modification timestamp on the file. See
+# https://www.gnu.org/software/make/manual/make.html#Empty-Targets
 	docker build -t $(REPO):$(TAG) .
+	touch $@
 
-push: container
-	docker push $(REPO):$(TAG)
+##############
+# Generating #
+##############
 
-clean:
-	rm -f $(BIN)
-	go clean -r $(MAIN_PKG)
-	docker images -q $(REPO) | xargs --no-run-if-empty docker rmi --force
-	rm -rf jsonnet/vendor
+.PHONY: vendor
```

Comment:
I am guessing we don't call this ever, I am fine with having this for consistency with other versions. But we still don't use gomods correctly?

Reply:
Argh, we need to disable go mods globaly here as we weren't using them in 4.1.